lib/middleware/route.js

```diff
@@ -69,13 +69,18 @@ exports.middleware = function route(next, app) {
     function addRoute(method, path, fn, name) {
         var keys = [];
         var spec = {keys: keys, fn: fn};
-        spec.pattern = path instanceof RegExp ? path : normalizePath(path, keys);
-        spec.weight = calcWeight(path);
+        if (path instanceof RegExp) {
```

Comment:
The Regexp stuff was there before... I didn't add it. I tried to use it and it didn't work, so I tried to fix it and it still doesn't really work right. Maybe I should just take these changes out, seeing as I'm not using them. If it is decided to allow reg exps for matching, this will all have to be rethought anyways...